lib/sse_client.ex

```diff
@@ -0,0 +1,65 @@
+defmodule MarathonEventExporter.SSEClient do
+  @moduledoc """
+  A GenServer to receive async responses from the HTTP client and forward the
+  data chunks to the SSE parser.
+  """
+
+  # The logging "functions" are actually macros, so we need to `require Logger`
+  # in order to access them.
+  require Logger
+
+  use GenServer
+
+  alias MarathonEventExporter.SSEParser
+
+  ## Client API
+
+  @doc """
+  Starts a new SSEClient.
+  """
+  def start({url, listeners, timeout}, opts \\ []) do
+    GenServer.start(__MODULE__, {url, listeners, timeout}, opts)
+  end
```

Comment:
We could default the timeout by defining a second clause like so:
`def start({url, listeners}, opts \\ []), do: start({url, listeners, 60_000}, opts)`

I'm not sure if that's necessarily a good idea, though.